packages/gateway/src/features/restLocation/locationHandler.ts

```diff
@@ -164,11 +164,31 @@ export async function createLocationHandler(
   const newLocation: fhir.Location = composeFhirLocation(payload)
   const partOfLocation = payload.partOf.split('/')[1]
 
-  const locations = await getLocationsByIdentifier(
-    `ADMIN_STRUCTURE_${String(payload.statisticalID)}`
-  ).catch((err) => {
-    throw err
-  })
+  let locations = []
+
+  locations.push(
+    ...(await getLocationsByIdentifier(
+      `${Code.ADMIN_STRUCTURE}_${String(payload.statisticalID)}`
+    ).catch((err) => {
+      throw err
+    }))
+  )
+
+  locations.push(
+    ...(await getLocationsByIdentifier(
+      `${Code.CRVS_OFFICE}_${String(payload.statisticalID)}`
+    ).catch((err) => {
+      throw err
+    }))
+  )
+
+  locations.push(
+    ...(await getLocationsByIdentifier(
+      `${Code.HEALTH_FACILITY}_${String(payload.statisticalID)}`
+    ).catch((err) => {
+      throw err
+    }))
+  )
```

Comment:
I'm not sure if we can use one GET req to do the search like `/location?code=HEALTH_FACILITY_...&code=CRVS_OFFICE_...` or `/location?code=HEALTH_FACILITY_...,=CRVS_OFFICE_...` but it's worth exploring. And if they don't work, then we can at least do a `Promise.all` to start the reqs in parallel

Reply:
converted to `Promise.all` for async request